arch/arm/include/cortex_m/tz.h

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief TrustZone API
+ *
+ * TrustZone API for Cortex-M23/M33 CPUs implementing the Security Extension.
+ */
+
+#ifndef _ARM_CORTEXM_TZ__H_
+#define _ARM_CORTEXM_TZ__H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifdef _ASMLANGUAGE
+
+/* nothing */
+
+#else
+
+#if defined(CONFIG_ARM_SECURE_FIRMWARE)
```

Comment:
I haven't seen this before.

Could you add a comment explaining why this ifdef is necessary?

Reply:
Will do. 
In principle, any TrustZone -related implementation concerns a Secure Firmware (i.e. a firmware that is meant to run in secure state)